website/source/docs/guides/deployment.html.md

```diff
@@ -0,0 +1,184 @@
+
+## 1.0 System Requirements
+
+Consul server agents are responsible for maintaining the cluster state, responding to RPC queries (read operations), and for processing all write operations. Given that Consul server agents do most of the heavy lifting, server sizing is critical for the overall performance efficiency and health of the Consul cluster.
+
+The following instance configurations are recommended.
+
+|Size|CPU|Memory|Disk|Typical Cloud Instance Types|
+|----|---|------|----|----------------------------|
+|Small|2 core|8-16 GB RAM|50 GB|**AWS**: m5.large, m5.xlarge|
+|||||**Azure**: Standard\_A4\_v2, Standard\_A8\_v2|
+|||||**GCE**: n1-standard-8, n1-standard-16|
+|Large|4-8 core|32-64+ GB RAM|100 GB|**AWS**: m5.2xlarge, m5.4xlarge|
+|||||**Azure**: Standard\_D4\_v3, Standard\_D5\_v3|
+|||||**GCE**: n1-standard-32, n1-standard-64|
+
+The **small size** instance configuration is appropriate for most initial production deployments, or for development/testing environments. The large size is for production environments where there is a consistently high workload.
+
+~> **NOTE** For high workloads, ensure that the disks support high IOPS to keep up with the high Raft log update rate.
+
+## 1.1 Datacenter Design
+
+A Consul cluster (typically 3 or 5 servers plus client agents) may be deployed in a single physical datacenter or it may span multiple datacenters. For a large cluster with high runtime reads and writes, deploying servers in the same physical location improves performance. In cloud environments, a single datacenter may be deployed across multiple availability zones i.e. each server in a separate availability zone within a single EC2 instance. Consul also supports multi-datacenter deployments via two separate clusters joined by WAN links. In some cases, one may also deploy two or more Consul clusters in the same LAN environment.
+
+### 1.1.1 Single Datacenter
+
+A single Consul cluster is recommended for applications deployed in the same datacenter. Consul supports traditional three-tier applications as well as microservices.
+
+Typically, there must be three or five servers to balance between availability and performance. These servers together run the Raft-driven consistent state store for catalog, session, prepared query, ACL, and KV updates.
+
+The recommended maximum cluster size for a single datacenter is `5,000` nodes. For a write-heavy and/or a read-heavy cluster, the maximum number of nodes may need to be reduced further, considering the impact of the number and the size of KV pairs and the number of watches. The time taken for gossip to converge increases as more client machines are added. Similarly, the time taken by the new server to join an existing multi-thousand node cluster with a large KV store and update rate may increase as they are replicated to the new server’s log.
+
+-> **TIP** For write-heavy clusters, consider scaling vertically with larger machine instances and lower latency storage.
```

Comment:
Format wise, Not sure TIP has much precedence in the existing docs, we may want to look around for other words. I see "Note" a lot.

Reply:
Fixed in 3c95575bc